drivers/gpu/drm/i915/intel_display.c

```diff
@@ -14187,6 +14201,13 @@ static void intel_setup_outputs(struct drm_device *dev)
 		}
 
 		intel_dsi_init(dev);
+
+		/* Force initialize DP on Port B */
```

Comment:
instead of adding a completely separate branch, can you instead modify the "if" condition that causes Port B to be probed on X540SA, to consider the DMI match?